### PR TITLE
Reset translation catalogue initialization

### PR DESCRIPTION
**Pull request by kingjia90**
related to PR #11374, more details in the review conversation

### CODE REVIEW

File: lib/Translation/Translator.php [hunk 1]
```diff
@@ -254,6 +254,15 @@ public function lazyInitialize($domain, $locale)
         }
     }
 
+    /**
+    * Resets the initialization of a specific catalogue 
+    */
+    public function resetInitialization($domain, $locale)
+    {
+        $cacheKey = 'translation_data_' . md5($domain . '_' . $locale);
```
Comment (dvesh3) [inline lib/Translation/Translator.php]: I think it is better to refactor cachekey generation to a separate method which then also called in `lazyInitialize()`

Comment (dvesh3) [inline lib/Translation/Translator.php:262]: I think it is better to refactor cachekey generation to a separate method which then also called in `lazyInitialize()`

Comment (dvesh3) [inline lib/Translation/Translator.php]: I think it is better to refactor cachekey generation to a separate method which then also called in `lazyInitialize()`

File: lib/Translation/Translator.php [hunk 2]
```diff
@@ -447,4 +459,15 @@ public function __call($method, $args)
...
         return call_user_func_array([$this->translator, $method], $args);
     }
+
+    /**
+     * @param string $domain
+     * @param string $locale
+     *
+     * @return string
+     */
+    private function getCacheKey($domain, $locale)
```
Comment (jdreesen) [inline lib/Translation/Translator.php]: Why not real types instead?

```suggestion
    private function getCacheKey(string $domain, string $locale): string
```

**Pull request merged**